sklearn/utils/validation.py

```diff
@@ -2169,17 +2175,18 @@ def _check_sample_weight(
         Validated sample weight. It is guaranteed to be "C" contiguous.
     """
     n_samples = _num_samples(X)
+    xp, _, device = get_namespace_and_device(X, sample_weight)
 
-    if dtype is not None and dtype not in [np.float32, np.float64]:
-        dtype = np.float64
+    if dtype is not None and dtype not in [xp.float32, xp.float64]:
+        dtype = _max_precision_float_dtype(xp, device)
```

Comment:
Can you update the description of the `dtype` parameter in the docstring? I think we should change the statement that `np.flaot64` will be the dtype of `sample_weight` if nothing else is specified. Maybe "The output will be the highest precision floating point dtype" or some such?

Reply:
I reworded this slightly to "Otherwise, the output has the highest precision floating point dtype supported by the array namespace/device of the input arrays."

Let me know if that sounds OK to you.